static/js/apps/eval_rig/feedback_form.tsx

```diff
@@ -14,181 +14,165 @@
  * limitations under the License.
  */
 
-import React, { useEffect, useState } from "react";
+import { doc, setDoc } from "firebase/firestore";
+import React, { useContext, useEffect, useState } from "react";
 
+import { db } from "../../utils/firebase";
+import { AppContext } from "./context";
+import { OneQuestion } from "./one_question";
+
+const initialResponse = {
+  overall: "",
+  question: "",
+  llmResponse: "",
+  dcResponse: "",
+  dcStat: "",
+};
 export interface EvalInfo {
   question: string;
   llmResponse: string;
   dcResponse: string;
   dcStat: string;
 }
 
+interface Response extends EvalInfo {
+  overall: string;
+  userEmail?: string;
+}
+
 export interface FeedbackFormProps {
+  queryId: string;
+  callId: string;
   evalInfo: EvalInfo;
 }
 
-const initialState = {
-  overall: "",
-  question: "",
-  llmResponse: "",
-  dcResponse: "",
-  dcStat: "",
-};
-
 export function FeedbackForm(props: FeedbackFormProps): JSX.Element {
-  const [responses, setResponses] = useState(initialState);
+  const { sheetId, userEmail } = useContext(AppContext);
+  const [response, setResponse] = useState<Response>(initialResponse);
 
   useEffect(() => {
-    // Reset responses whenever evalInfo changes
-    setResponses(initialState);
+    setResponse(initialResponse);
   }, [props.evalInfo]);
 
   const handleChange = (event) => {
     const { name, value } = event.target;
-    setResponses((prevState) => ({
+    setResponse((prevState) => ({
       ...prevState,
       [name]: value,
     }));
   };
 
   const handleSubmit = (event) => {
     event.preventDefault();
-    saveResponses(responses);
+    response.userEmail = userEmail;
+    saveResponse(sheetId, props.queryId, props.callId, response);
   };
 
   return (
     <form onSubmit={handleSubmit}>
       <fieldset>
         <div>
           <h2>OVERALL EVALUATION</h2>
-          <h3>Are there any hallucinations?</h3>
-          <label>
-            <input
-              type="radio"
-              name="overall"
-              value="0"
-              checked={responses.overall === "0"}
-              onChange={handleChange}
-            />
-            Yes
-          </label>
-          <label>
-            <input
-              type="radio"
-              name="overall"
-              value="1"
-              checked={responses.overall === "1"}
-              onChange={handleChange}
-            />
-            No
-          </label>
+          <OneQuestion
+            question="Are there any hallucinations?"
```

Comment:
nit:  "How is the overall answer?"  (since Prem didn't want to call out hallucination)

Reply:
Done